### ISSUE TITLE
Val loss log

### ISSUE THREAD

**Issue by zhengyang-wang** [facebookresearch/lingua#29]
The current eval implementation does not seem to support loss computation on a validation set during eval.
[eval code](https://github.com/facebookresearch/lingua/blob/main/apps/main/eval.py#L181)

Is there any plan to add the support?

**Comment by BadrYoubiIdrissi** [comment]
Yes we will very probably be migrating to lighteval instead of lm-eval harness as it offers overall more evals. We would love a PR on this if you're interested.

**Comment by narain1** [comment]
I'd be interested in contributing a PR for this migration. Could you please provide more details on the scope and requirements? Specifically, is the goal to fully transition all evaluations to LightEval, or are there specific parts that need prioritization?

**Comment by Hprairie** [comment]
+1 being able to calculate perplexity on validation sets would be really nice. Automatically detecting the subsets from the `sources` of the data config, and evaluating perplexity on the `*.val.*.jsonl` chunks in the subfolder would be amazing.

**Comment by mathuvu** [comment]
This is now being tested in PR #47
